File: 0567-permutation-in-string/0567-permutation-in-string.cpp
class Solution {
public:
    bool checkInclusion(string s1, string s2) {
        vector<int> v1(26, 0);
        
        for(auto i:s1)
            v1[i-'a']++;
        
        int n1=s1.size(), n2=s2.size();
        
        for(int i=0; i<=n2-n1; i++)
        {
            vector<int>v2(26, 0);
            for(int j=i; j<n1+i; j++)
                v2[s2[j]-'a']++;
            
            if(v2 == v1)
                return true;
        }
        return false;
    }
};

File: 0567-permutation-in-string/0567-permutation-in-string.py
class Solution:
    def usingCounter(self, s1, s2, n1, n2):
        c1 = Counter(s1)
        
        for i in range(n2-n1+1):
            c2 = Counter(s2[i:n1+i])
            
            if c1 == c2:
                return True
        return False
        
    def checkInclusion(self, s1: str, s2: str) -> bool:
        n1, n2 = len(s1), len(s2)
        
        return self.usingCounter(s1, s2, n1, n2)